desktop/src/hooks/useFolderList/useFolderList.ts

```diff
@@ -0,0 +1,216 @@
+        "fdDate": "2024/12/01"
+      }
+    },
+    "files": {
+      "D:/test/Folder1": [
+        {
+          "filepath": "/path/to/folder1/file1.txt",
+          "filename": "file1.txt",
+          "size": "12MB",
+          "birthtime": "2024-11-01T10:00:00Z",
+          "lastModified": "2024-11-10T12:00:00Z",
+          "lastAccessed": "2024-11-15T15:00:00Z",
+          "checksum": "abcd1234"
+        },
+        {
+          "filepath": "/path/to/folder1/file2.txt",
+          "filename": "file2.txt",
+          "size": "2MB",
+          "birthtime": "2024-11-02T10:00:00Z",
+          "lastModified": "2024-11-11T12:00:00Z",
+          "lastAccessed": "2024-11-16T15:00:00Z",
+          "checksum": "efgh5678"
+        }
+      ],
+      "D:/test/Folder2": [
+        {
+          "filepath": "/path/to/folder2/file3.txt",
+          "filename": "file3.txt",
+          "size": "3MB",
+          "birthtime": "2024-11-03T10:00:00Z",
+          "lastModified": "2024-11-12T12:00:00Z",
+          "lastAccessed": "2024-11-17T15:00:00Z",
+          "checksum": "ijkl91011"
+        }
+      ]
+    },
+    "admin": {
+      "accession": "1234567",
+      "application": "7654321"
+    }
+  }
+
+}
+
+const testToken = 'eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJxUWlFWDB2T2Z1SlBuWUw4MWo0Q2tDOHVPdEJ1aFZvM0xBd2ppczZWbHRzIn0.eyJleHAiOjE3Mzc0MTA3ODMsImlhdCI6MTczNzQxMDQ4MywiYXV0aF90aW1lIjoxNzM3NDEwNDgzLCJqdGkiOiIzZGYxNzg4Yi05MzE2LTQzZmUtYmI1ZC02ZTlmNmYxN2JkMTMiLCJpc3MiOiJodHRwczovL2Rldi5sb2dpbnByb3h5Lmdvdi5iYy5jYS9hdXRoL3JlYWxtcy9zdGFuZGFyZCIsImF1ZCI6ImNpdHotZ3JzLWRhdHMtY2lybW8taW1iLTU2MjMiLCJzdWIiOiJhMzJkNmY4NTljNjY0NTBjYTQ5OTViMGIyYmYwYTg0NEBpZGlyIiwidHlwIjoiQmVhcmVyIiwiYXpwIjoiY2l0ei1ncnMtZGF0cy1jaXJtby1pbWItNTYyMyIsInNlc3Npb25fc3RhdGUiOiI4ZTRhMzVlMC01ODFhLTRkNTktYjE2OC03ZDhjNDlkODA4ZGEiLCJzY29wZSI6Im9wZW5pZCBpZGlyIGVtYWlsIHByb2ZpbGUgYXp1cmVpZGlyIiwic2lkIjoiOGU0YTM1ZTAtNTgxYS00ZDU5LWIxNjgtN2Q4YzQ5ZDgwOGRhIiwiaWRpcl91c2VyX2d1aWQiOiJBMzJENkY4NTlDNjY0NTBDQTQ5OTVCMEIyQkYwQTg0NCIsImlkZW50aXR5X3Byb3ZpZGVyIjoiaWRpciIsImlkaXJfdXNlcm5hbWUiOiJTVE9FV1MiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJUb2V3cywgU2NvdHQgRCBDSVRaOkVYIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiYTMyZDZmODU5YzY2NDUwY2E0OTk1YjBiMmJmMGE4NDRAaWRpciIsImRpc3BsYXlfbmFtZSI6IlRvZXdzLCBTY290dCBEIENJVFo6RVgiLCJnaXZlbl9uYW1lIjoiU2NvdHQiLCJmYW1pbHlfbmFtZSI6IlRvZXdzIiwiZW1haWwiOiJzY290dC50b2V3c0Bnb3YuYmMuY2EifQ.BHKiTmqpWHSU8BUZMzo5EIkyqKFbQFxFaZc-trD6iVG7IWnxZFDTqVPPMvNXiCqQrx7prSDTXAZrAxZqA_UwAsYCsf2lL7kE82QOXGiECVuRFtAF-yn0iTmNEFR61dgyJDnrOy5fMZllLum0hiuxydi_OpSGflQgNgtm0PFwQ2zlyY2OWm6XptIMSauSGo-mIsgav4cRpA8atWiDsVf_lrHK6xBSYv1qTQajsjmDcsUNGq83F3QGaZ8aSxxfOBj8uZNgfSquLx6HxX3-F-RbT6-bRZ9op4Y422RjXxFNW6TEOmB6dPXiEwtOjR4MfIntcv-JMeBNQmXjvOujv7iNHg'
+
+export const useFolderList = ({ accessToken }) => {
+  const [folders, setFolders] = useState<FolderRow[]>([])
+  const [metaData, setMetaData] = useState<Record<string, unknown>>({})
+  const [workers] = useState(window.api.workers);
```

Comment:
There is a type error with window.api. I suspect it to be because hooks falls outside the structure of `main`, `preload`, `renderer`